crates/cadenza-syntax/STATUS.md

```diff
@@ -0,0 +1,450 @@
+# Cadenza Syntax Parser - Status Document
+
+This document tracks the current state of the `cadenza-syntax` crate and remaining parser implementation work.
+
+## Overview
+
+The syntax crate implements a whitespace-significant, keyword-less parser for Cadenza using Pratt parsing with binding powers. The parser produces a lossless CST (Concrete Syntax Tree) using the rowan library.
+
+## Current State
+
+The parser implements most core syntax features:
+
+✅ **Completed:**
+- Basic lexer and parser infrastructure
+- All operator categories (arithmetic, comparison, logical, bitwise, shift, assignment, range)
+- Field access (`.`) and path access (`::`)
+- Prefix operators (`@`, `!`, `~`, `$`, `...`)
+- Postfix operators (`?`, `|?`)
+- Array literals (`[1, 2, 3]`)
+- Record literals - **shorthand only** (`{ x, y }`)
+- Parenthesized expressions
+- Indentation-based blocks
+- Apply nodes for function application
+- Whitespace significance tracking
+- Generated binding power system
+
+⚠️ **Partially Complete:**
+- **Records**: Shorthand syntax works (`{ x, y }`), but field assignments (`{ x = 1 }`) fail due to parser marker propagation issue
```

Comment:
This should be working now. Make sure to look through all of the `test-data` files and their corresponding snapshot files.